--- original/opam	2020-05-21 06:32:45.499928663 +0900
+++ opam.modified	2020-05-21 07:57:18.240589323 +0900
@@ -12,8 +12,8 @@ depends: [
   "ctypes" {>= "0.4"}
   "ounit" {with-test}
   "ocamlfind" {build}
-  "conf-llvm" {build & = version}
-  "conf-python-2-7" {build}
+  "conf-llvm" {build & = "10.0.0"}
+  "conf-python-3" {build}
   "conf-cmake" {build}
 ]
 install: [
@@ -28,9 +28,9 @@ install: [
 dev-repo: "git+http://llvm.org/git/llvm.git"
 extra-files: [
   ["link-META.patch" "md5=ef4ebb8706be2ed402f31fc351d7dc75"]
-  ["install.sh" "md5=c8122f1f3c974594ba6c8767b187cdab"]
+  ["install.sh" "md5=fe5d42ced8b970c7cf44a755bb5ad179"]
   ["fix-shared.patch" "md5=dce86b1db352332968ceb6d042b408a8"]
-  ["META.patch" "md5=1d0af08bab7a0f831f68849b6556e414"]
+  ["META.patch" "md5=8a6ee82d643e1d424ff06477a2bd5a39"]
 ]
 url {
   src:
